src/librustc/ty/mod.rs

```diff
@@ -2623,3 +2575,47 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         }
     }
 }
+
+fn associated_item<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId)
```

Comment:
I think this should be in `rustc::hir`, including `associated_item_from_{impl,trait}_item_ref`.